js/src/edit-block/hooks/useField.js

```diff
@@ -223,6 +224,26 @@ const useField = () => {
 		editPost( { content: JSON.stringify( fullBlock ) } );
 	}, [ blockNameWithNameSpace, editPost, fullBlock ] );
 
+	/**
+	 * Duplicates this field.
+	 */
+	const duplicateField = useCallback( ( fieldName ) => {
+		const currentField = getField( fieldName );
+		const { fields = {} } = block;
+		const newFieldNumber = Object.values( fields ).length + 1;
+
+		const newFieldName = `duplicated-field-${ newFieldNumber.toString() }`;
```

Comment:
When duplicating a field name of `foo-field`, its duplicates would be `foo-field-1`, `foo-field-2`, `foo-field-3`.

Feel free to modify `getNewFieldNumber` and call it Maybe it could accept a second parameter of `baseFieldName` or something.

For example, for a field

Reply:
Done in the last commit :)